vpr/src/base/atom_netlist_utils.cpp

```diff
@@ -870,6 +879,11 @@ bool remove_buffer_lut(AtomNetlist& netlist, AtomBlockId blk, int verbosity) {
     } else if (!(driver_is_pi || po_in_input_sinks) && po_in_output_sinks) {
         //Must use the output name to perserve primary-output name
         new_net_name = netlist.net_name(output_net);
+    } else if (new_driver_is_clock) {
+        auto input_net_name = netlist.net_name(input_net);
+        auto output_net_name = netlist.net_name(output_net);
+        new_net_name = input_net_name;
+        netlist.add_clock_net_alias(output_net_name, input_net_name);
```

Comment:
This branch would no longer be needed, and calls to:
```cpp
netlist.add_net_alias(new_net_name, input_net_name);
netlist.add_net_alias(new_net_name, output_net_name);
```
would be always added.

Reply:
Done, also added a check during insertion to see if the alias key is already present and if so, the content needs to match to the net that was to be assigned.